mache-couchbase/src/main/java/com/excelian/mache/couchbase/builder/CouchbaseProvisioner.java

```diff
@@ -0,0 +1,40 @@
+package com.excelian.mache.couchbase.builder;
+
+import com.excelian.mache.builder.StorageProvisioner;
+import com.excelian.mache.core.Mache;
+import com.excelian.mache.core.MacheFactory;
+import com.excelian.mache.core.SchemaOptions;
+import com.excelian.mache.couchbase.CouchbaseCacheLoader;
+import com.excelian.mache.couchbase.CouchbaseConfig;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * {@link StorageProvisioner} implementation for Couchbase.
+ */
+public class CouchbaseProvisioner implements StorageProvisioner {
+
+    @Override
+    public String getStorage() {
+        return "Couchbase";
+    }
+
+    @Override
+    public <K, V> Mache<K, V> getCache(String keySpace, Class<V> valueType, SchemaOptions schemaOption,
+                                       ClusterDetails clusterDetails, StorageServerDetails... serverDetails) {
+
+        List<String> addresses = Arrays.stream(serverDetails)
+                .map(StorageServerDetails::getAddress)
+                .collect(Collectors.toList());
+
+        CouchbaseConfig config = CouchbaseConfig.builder()
+                .withCacheType(valueType)
+                .withServerAddresses(addresses)
+                .withBucketName(keySpace)
+                .build();
```

Comment:
I think that may then make a compiletime dependency from the builder on the individual modules, which would mean you'd need to drag the mache-cassandra jar with you if you want to build a mache backed by couch, because the builder class won't be able to be loaded without all its dependent classes that are now housed in the different backing store components.  This is why I use strings or delegation in the XXXProvisioners....I'm up for exploring it further with you, but the main requirement would be to keep the jar dependencies as separate as possible.